tests/api2/test_bootenv.py

```diff
@@ -1,24 +1,61 @@
+from functions import wait_on_job
 from middlewared.test.integration.utils import call, ssh
 
+def test_get_default_environment_and_make_new_one():
+    active_be_id = call('bootenv.query', [['activated', '=', True]], {'get': True})['id']
+
+    # create duplicate name to test failure
+    results = call('bootenv.create', {'name': active_be_id, 'source': active_be_id})
+    assert results == f'[EEXIST] bootenv_create.name: The name "{active_be_id}" already exists'
```

Comment:
I would have expected the call to throw an exception and therefore need to be something _along the lines of_:
```
    with pytest.raises(ValidationErrors) as ve:
        call('bootenv.create', {'name': active_be_id, 'source': active_be_id})
    assert ve.value.errors == [
        ValidationError(
            'bootenv_create.name',
            f'The name "{active_be_id}" already exists',
            errno.EEXIST
        )]
```